jib-cli/src/test/java/com/google/cloud/tools/jib/cli/buildfile/BuildFileSpecTest.java

```diff
@@ -169,146 +169,112 @@ public void testBuildFileSpec_nullCollections() throws JsonProcessingException {
     assertThat(parsed.getCmd()).isEmpty();
   }
 
-  @RunWith(Parameterized.class)
-  public static class OptionalStringCollectionTests {
-
-    @Parameterized.Parameters(name = "{0}")
-    public static Collection<Object[]> data() {
-      return Arrays.asList(new Object[][] {{"volumes"}, {"exposedPorts"}, {"entrypoint"}, {"cmd"}});
-    }
-
-    @Parameterized.Parameter public String fieldName;
-
-    @Test
-    public void testBuildFileSpec_noNullEntries() {
-      String data =
-          "apiVersion: v1alpha1\n" + "kind: BuildFile\n" + fieldName + ": ['first', null]";
-
-      JsonProcessingException exception =
-          assertThrows(
-              JsonProcessingException.class, () -> mapper.readValue(data, BuildFileSpec.class));
-      assertThat(exception)
-          .hasMessageThat()
-          .contains("Property '" + fieldName + "' cannot contain null entries");
-    }
-
-    @Test
-    public void testBuildFileSpec_noEmptyEntries() {
-      String data = "apiVersion: v1alpha1\n" + "kind: BuildFile\n" + fieldName + ": ['first', ' ']";
-
-      JsonProcessingException exception =
-          assertThrows(
-              JsonProcessingException.class, () -> mapper.readValue(data, BuildFileSpec.class));
-      assertThat(exception)
-          .hasMessageThat()
-          .contains("Property '" + fieldName + "' cannot contain empty strings");
-    }
-
-    @Test
-    public void testBuildFileSpec_emptyOkay() throws JsonProcessingException {
-      String data = "apiVersion: v1alpha1\n" + "kind: BuildFile\n" + fieldName + ": []";
-
-      assertThat(mapper.readValue(data, BuildFileSpec.class)).isNotNull();
-    }
-
-    @Test
-    public void testBuildFileSpec_nullOkay() throws JsonProcessingException {
-      String data = "apiVersion: v1alpha1\n" + "kind: BuildFile\n" + fieldName + ": null";
-
-      assertThat(mapper.readValue(data, BuildFileSpec.class)).isNotNull();
-    }
+  @Test
+  @Parameters(value = {"volumes", "exposedPorts", "entrypoint", "cmd"})
+  public void testBuildFileSpec_noNullEntries(String fieldName) {
```

Comment:
Oops sorry I missed this. Perhaps we can combine this test with the one below. Similar to what sonarcloud [suggested](https://sonarcloud.io/project/issues?id=GoogleContainerTools_jib&issues=AX0Fd6hL2M0IIkbbKH6o&open=AX0Fd6hL2M0IIkbbKH6o&pullRequest=3496)

Reply:
They are testing different things and their error messages are different. I think merging them into one is too extreme. The current granularity seems good.

Reply:
Sounds good.